RealmSwift/Query.swift

```diff
@@ -0,0 +1,856 @@
+                        arguments.append(high)
+                    } else {
+                        throwRealmException("Could not construct .contains(_:) predicate")
+                    }
+                case let .contains(val):
+                    predicateString.insert("%@ IN ", at: predicateString.count-1)
+                    arguments.append(val.objCValue)
+                case let .containsAny(col):
+                    predicateString.append(" IN %@")
+                    arguments.append(col)
+                }
+            case let .compound(comp):
+                predicateString.append(" \(comp.rawValue) ")
+            case let .keyPath(name, isCollection):
+                // For the non verbose subqery
+                if isCollection && isSubquery {
+                    predicateString.append("$obj")
+                    continue
+                }
+                // Anything below the verbose subquery uses
+                var needsDot = false
+                if idx > 0, case .keyPath = tokens[idx-1] {
+                    needsDot = true
+                }
+                if needsDot {
+                    predicateString.append(".")
+                }
+                predicateString.append("\(name)")
+            case let .stringSearch(s):
+                switch s {
+                case let .contains(str, options):
+                    predicateString.append(" CONTAINS\(optionsStr(options)) %@")
+                    arguments.append(str)
+                case let .like(str, options):
+                    predicateString.append(" LIKE\(optionsStr(options)) %@")
+                    arguments.append(str)
+                case let .beginsWith(str, options):
+                    predicateString.append(" BEGINSWITH\(optionsStr(options)) %@")
+                    arguments.append(str)
+                case let .endsWith(str, options):
+                    predicateString.append(" ENDSWITH\(optionsStr(options)) %@")
+                    arguments.append(str)
+                case let .equals(str, options):
+                    predicateString.append(" ==\(optionsStr(options)) %@")
+                    arguments.append(str)
+                case let .notEquals(str, options):
+                    predicateString.append(" !=\(optionsStr(options)) %@")
+                    arguments.append(str)
+                }
+            case let .rhs(v):
+                predicateString.append(" %@")
+                arguments.append(v.objCValue)
+            case let .subquery(col, str, args):
+                predicateString.append("SUBQUERY(\(col), $obj, \(str)).@count")
+                arguments.append(contentsOf: args)
+            case let .collectionAggregation(agg):
+                predicateString.append(agg.rawValue)
+            case let .keypathCollectionAggregation(agg):
+                predicateString.insert(agg.rawValue, at: predicateString.count-2)
```

Comment:
@dianaafanador3 
This will crash if the query expression only contains one keyPath. e.g this test will crash:
```swift
assertCollectionQuery(on: \.intCol,
                              isList: true,
                              predicate: "intCol.@avg > %@",
                              values: [1],
                              expectedCount: 1) {
            $0.intCol.avg > 1
        }
```